src/sqlfluff/rules/L041.py

```diff
@@ -35,21 +35,36 @@ class Rule_L041(BaseRule):
     def _eval(self, context: RuleContext) -> Optional[LintResult]:
         """Select clause modifiers must appear on same line as SELECT."""
         if context.segment.is_type("select_clause"):
+            # Scan the select clause's children. Look for three things in sequence:
```

Comment:
The previous version of this rule would get confused by newlines in unexpected places (i.e. **before** the `SELECT`), so I replaced the three individual segment searches with this loop, which **only** triggers if there's a newline **between** `SELECT` and `DISTINCT`.